common/src/chain/config/mod.rs

```diff
@@ -316,6 +316,24 @@ impl ChainConfig {
         }
     }
 
+    pub fn spend_share_maturity_distance(&self, block_height: BlockHeight) -> BlockDistance {
+        match self.net_upgrades.consensus_status(block_height) {
+            RequiredConsensus::IgnoreConsensus | RequiredConsensus::PoW(_) => {
+                self.empty_consensus_reward_maturity_distance
+            }
+            RequiredConsensus::PoS(status) => {
+                let pos_config = match &status {
+                    PoSStatus::Ongoing(config)
+                    | PoSStatus::Threshold {
+                        initial_difficulty: _,
+                        config,
+                    } => config,
+                };
+                pos_config.spend_share_maturity_distance()
+            }
+        }
```

Comment:
If you need to specialize for tests, please change ChainConfig in the tests to reflect the values you want through the builder. Having special rules here is a bad idea. We reserve this for implementations or specific chain specifications.

Reply:
This code specializes for consensus type and is not test-specific per se. Specific values can be set by providing PoSChainConfig struct when constructing ChainConfig if necessary.

P.S. I cleaned up this code a bit take a look at newer version